src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj

```diff
@@ -2,12 +2,17 @@
 
   <PropertyGroup>
     <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
-    <TargetFrameworks>netstandard2.0;net462</TargetFrameworks>
+    <TargetFrameworks>net6.0;net462</TargetFrameworks>
```

Comment:
@alanwest @utpilla @cijothomas Hey I dropped `netstandard2.0` and added `net6.0`. The `netstandard2.0` didn't really make sense to me. We've seen users reference this in `netstandard2.0` projects which then blow up when consumed into `net462` projects because the API is different. Seems like this is a _safer_ way to go about things?

Reply:
This would resolve #3434. Technically, this is sort of a breaking change.

I suspect that the use of the HttpClient instrumentation is pretty ubiquitous. Given that this change will require .NET Framework users who have shared netstandard libraries to retarget their libraries, I think we may as well rip the bandaid off and remove netstandard targets from all our packages.

Reply:
agree with this change to HttpClientInstrumenttion.
Those consuming this via shared library will see breakage, unless the shared library also updates their target.
Those consuming this directly from app (like a Asp.Net or Asp.Net core) app should see no breakage.

Changing ns20  target from API/SDK packages should be discussed separately, as those are stable packages, unlike httpclient instrumentation.

Reply:
I merged this. Felt like it was good to get it in, get it out there, see if we get any feedback.

This change would have also worked:

`<TargetFrameworks>netstandard2.1;net462</TargetFrameworks>`

Because .NET Framework does not implement .NET Standard 2.1, users would get the correct version in that case.

Not sure which is the best way to go.

* We don't have `netstandard2.1` in AspNetCore instrumentation currently.
* We do have `netstandard2.0` in API+SDK and `netstandard2.1` in SDK. 

If our goal is to drop support for <net6.0 we should maybe just go to `<TargetFrameworks>net6.0;net462</TargetFrameworks>` across the board. It will be bumpy, but clear. Otherwise perhaps we should keep `netstandard2.1` in HttpClient + AspNetCore.